### DISCUSSION TITLE
Currency symbol before or after price

### DISCUSSION THREAD

**Discussion by jouniikaheimo** [filamentphp/filament#45] [General]
Currency symbol choice before or after ?

**Comment by ManojKiranA** [comment]
In `form fileds` or  in `tables` while displaying

**Comment by ewald1976** [comment]
You can do something like this:

`Tables\Columns\TextColumn::make('price')->getStateUsing(fn ($record): string =>
                              money($record->price, YOUR_CURRENCY))`

Before or after the value depends on the currency afaik. For example € after $ before.

**Reply by danharrin** [reply]
`TextColumn::make('price')->money('YOUR_CURRENCY')`

**Reply by mikailhamzat** [reply]
Some currencies in Africa don't display. Is there a way to add a custom currency symbol? Thanks

**Reply by danharrin** [reply]
->formatStateUsing() allows you to transform the state in any way

**Reply by clementmas** [reply]
`TextColumn::make('price')->money('EUR')` displays €10.00 instead of 10.00€

Is using `formatStateUsing` on every money column the only solution?

**Reply by danharrin** [reply]
We are just using Laravels formatting method, if they add an option for symbol position then we will add support for it too.

**Reply by danharrin** [reply]
I read the docs for you. If you pass `locale: 'de'`, it formats it after:

https://laravel.com/docs/11.x/helpers#method-number-currency

**Reply by clementmas** [reply]
Thanks Dan. It's indeed an issue with the underlying `(new NumberFormatter(app()->getLocale(), NumberFormatter::CURRENCY))->formatCurrency(12.45, 'EUR')` which dispays the € sign before unless we manually specify `fr` or `de` as a locale.

We also can't control the number of decimals to display so I'll use a custom macro instead.

**Reply by danharrin** [reply]
I don't think it's an issue with the number formatter, as far as I know some EUR locales add the symbol before

**Comment by clementmas** [comment]
I'm also looking for a way to display the EUR € symbol after when using English as the default language.
